### PR TITLE
Add documentation for build environment variables

### PR DESCRIPTION
**Pull request by sethvargo**

### CODE REVIEW

File: packer/builds/build-environment.html.md [hunk 1]
```diff
@@ -67,3 +67,19 @@ the following hardware resources. This is subject to change.
...
 
 If this doesn't suit your needs, please [contact us](mailto:support@hashicorp.com).
+
+### Environment Variables
+
+During each Packer build, the following environment variables are available as
+part of the build:
+
+- `ATLAS_TOKEN` - a unique token that can communicate with Atlas for the
+duration of this build. This token will expire upon completion of the build.
```
Comment (pearkes) [inline packer/builds/build-environment.html.md]: I'd add a note here about how this is used as part of any atlas providers or post-processors.

Comment (pearkes) [inline packer/builds/build-environment.html.md]: (implicitly, but just call attention to that)

File: packer/builds/build-environment.html.md [hunk 2]
```diff
@@ -67,3 +67,19 @@ the following hardware resources. This is subject to change.
...
+- `ATLAS_JOB_NUMBER` - a unique identifier for the complete Packer run including
+all builds. Given a Packer template with multiple builders, all builders will be
+grouped under a single "job number".
+- `ATLAS_BUILD_ID` - a unique identifier for this particular builder within a
+job. This is an internal number that uniquely refers to the particular builder
+in a job.
+- `ATLAS_BUILD_CONFIGURATION_VERSION` - the version of the build configuration
```
Comment (pearkes) [inline packer/builds/build-environment.html.md]: You can link to the glossary here I think.

File: packer/builds/build-environment.html.md [hunk 3]
```diff
@@ -67,3 +67,20 @@ the following hardware resources. This is subject to change.
...
+During each Packer build, the following environment variables are available as
+part of the build:
+
+- `ATLAS_TOKEN` - a unique token that can communicate with Atlas for the
+duration of this build. This token will expire upon completion of the build.
+This token is used as part of any Atlas-specific providers or post processors.
+- `ATLAS_JOB_NUMBER` - a unique identifier for the complete Packer run including
+all builds. Given a Packer template with multiple builders, all builders will be
+grouped under a single "job number".
+- `ATLAS_BUILD_ID` - a unique identifier for this particular builder within a
```
Comment (pearkes) [inline packer/builds/build-environment.html.md]: It doesn't look like we export this. We probably don't need to, either.

File: packer/builds/build-environment.html.md (hunk 1 above)
Comment (sethvargo) [inline packer/builds/build-environment.html.md]: Added

**Pull request merged**